### PR TITLE
feat: Add Attached Sheet to Playground

### PR DESCRIPTION
**Pull request by ABausG**
## Description

Adding an AttachedFloatingBottomSheetType to the Playground App.
This works by using a `GlobalKey` as the anchor to Align the Sheet at the Position of the Anchor.

One thing I think can still be improved is the animation. Ideally it would be nice if the Modal Scales out from the anchor position (+ Transformation based on Anchor Size/Alignment)

A Demo (before improvements to the Alignments/Size Constraints can be found here:

https://github.com/user-attachments/assets/9b1f6b7c-4a64-46c8-9eae-91cce0d5893d

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors.
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] The package compiles with the minimum Flutter version stated in the [pubspec.yaml](https://github.com/woltapp/wolt_modal_sheet/blob/main/pubspec.yaml#L8)

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[Contributor Guide]: https://github.com/woltapp/wolt_modal_sheet/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: playground/lib/home/custom_sheets/attached_floating_bottom_sheet_type.dart [hunk 1]
```diff
@@ -0,0 +1,169 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:wolt_modal_sheet/wolt_modal_sheet.dart';
+
+class AttachedFloatingBottomSheetType extends WoltModalType {
```
Comment (ulusoyca) [inline playground/lib/home/custom_sheets/attached_floating_bottom_sheet_type.dart:6]: Can you ask ChatGpt to add extensive code comment about what this modal type does? What does attaching mean, and what the properties do? You can also explain your own logic.

File: playground/lib/home/home_screen.dart [hunk 2]
```diff
@@ -125,6 +145,25 @@ class _HomeScreenState extends State<HomeScreen> {
...
+                            onPressed: () {
+                              WoltModalSheet.show(
+                                context: context,
+                                modalTypeBuilder: (_) =>
+                                    AttachedFloatingBottomSheetType(
+                                  anchorKey: _attachedRandomKey,
+                                  alignment: Alignment.bottomCenter,
+                                ),
+                                pageListBuilder: (_) =>
+                                    [AdjustTimeNotificationPage()],
```
Comment (ulusoyca) [inline playground/lib/home/home_screen.dart:162]: Can you add an option to hide the back button in this page? Maybe adding a boolean field `isFirstPage` and setting it true by default, but setting it false in this case, then hide the back button accordingly?

image

**Review by ulusoyca — CHANGES_REQUESTED** [review]

**Comment by ulusoyca** [comment]
Thank you very much for the example!

**Comment by ulusoyca** [comment]
@ABausG When RTL is selected, there is weird behavior. Can you check?

<video src="https://github.com/user-attachments/assets/098ac0fd-a434-48c3-9d21-8fb8ff9de435">

File: playground/lib/home/custom_sheets/attached_floating_bottom_sheet_type.dart (hunk 1 above)
Comment (ABausG) [reply playground/lib/home/custom_sheets/attached_floating_bottom_sheet_type.dart:31]: Added some hybrid doc comments. Some custom written some by Copilot

File: playground/lib/home/home_screen.dart (hunk 2 above)
Comment (ABausG) [reply playground/lib/home/home_screen.dart]: Exchanged this for the modal used elsewhere in the playground

**Comment by ABausG** [comment]
> @ABausG When RTL is selected, there is weird behavior. Can you check?
> 

Did not test that!! Solved by swapping the passed alignment based on the directionality. Works for me now!

File: playground/lib/home/custom_sheets/attached_floating_bottom_sheet_type.dart (hunk 1 above)
Comment (ulusoyca) [reply playground/lib/home/custom_sheets/attached_floating_bottom_sheet_type.dart:31]: Simply amazing! Thanks

**Comment by ulusoyca** [comment]
👑 AMAZING! Super happy <3

**Review by ulusoyca — APPROVED** [review]

File: playground/lib/home/home_screen.dart [hunk 3]
```diff
@@ -1,8 +1,10 @@
 import 'package:demo_ui_components/demo_ui_components.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/scheduler.dart';
+import 'package:playground/home/custom_sheets/attached_floating_bottom_sheet_type.dart';
 import 'package:playground/home/custom_sheets/floating_bottom_sheet_type.dart';
 import 'package:playground/home/custom_sheets/top_notification_sheet_type.dart';
+import 'package:playground/home/pages/custom_sheet_pages/adjust_time_notification_page.dart';
```
Comment (durannumit) [inline playground/lib/home/home_screen.dart:7]: we are not using that import, would be nice to remove 🚀

**Review by durannumit — CHANGES_REQUESTED** [review]
- still there are some issues on mobile

| LTR | RTL |
|--------|--------|
| Screenshot_1721330505 | Screenshot_1721330513 |

For me looks really good on web btw

**Comment by ABausG** [comment]
Did a further refactor to now auto-calculate the Alignment that results in the most available space. By my testing (resizing to super small form factors) this now solves the issue as in the screenshot from mobile

**Comment by durannumit** [comment]
> Did a further refactor to now auto-calculate the Alignment that results in the most available space. By my testing (resizing to super small form factors) this now solves the issue as in the screenshot from mobile

I tried on latest version, it looks same. Let me test on more devices also

**Comment by ABausG** [comment]
@durannumit I did some more adjustments to the maxWidth calculation but for me it also works fine on iOS and Android Simulator 🤔

**Comment by durannumit** [comment]
> @durannumit I did some more adjustments to the maxWidth calculation but for me it also works fine on iOS and Android Simulator 🤔

it works well now! Thanks for quick adjustment, LGTM 🚀

**Review by durannumit — APPROVED** [review]

**Pull request merged**